torch/package/package_exporter.py

```diff
@@ -670,6 +670,7 @@ def _check_mocked_error(module: Optional[str], field: Optional[str]):
                 if pickle_protocol == 4:
                     if (
                         opcode.name == "SHORT_BINUNICODE"
+                        or opcode.name == "BINUNICODE"
```

Comment:
I am still perplexed. Don't we have to do the BINUNICODE special handling for older pickle protocol too?

Reply:
Ah, no: this was actually a change in between 3 & 4
https://peps.python.org/pep-3154/#serializing-more-lookupable-objects

STACK_GLOBAL was added as part of 4 that takes the topmost stack items to figure out the values and looks up the strings  as module_name and qual_name (so this code basically keeps a running buffer of the last 2 strings seen just in case it runs into a STACK_GLOBAL)

In pickle 3 there was just GLOBAL that accepts the module & field as an argument -- so no separate string handling/no custom binunicodes, handled in the if/else around line 700.
https://github.com/python/cpython/blob/main/Lib/pickletools.py#L1928

Reply:
ok!